dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPIImpl.java

```diff
@@ -104,17 +126,30 @@ public FileAsset checkinFile(Contentlet fileCon, User user,boolean respectFronte
 	}
 	 */
 	@CloseDBIfOpened
-	public List<FileAsset> findFileAssetsByFolder(Folder parentFolder, User user, boolean respectFrontendRoles) throws DotDataException,
-			DotSecurityException {
-		List<FileAsset> assets = null;
+	public List<FileAsset> findFileAssetsByFolder(
+			final Folder parentFolder,
+			final User user,
+			final boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
+		List<Contentlet> contentlets = null;
+
 		try{
-			assets = fromContentlets(perAPI.filterCollection(contAPI.search("+structureType:" + Structure.STRUCTURE_TYPE_FILEASSET+" +conFolder:" + parentFolder.getInode(), -1, 0, null , user, respectFrontendRoles),
-					PermissionAPI.PERMISSION_READ, respectFrontendRoles, user));
+			contentlets = contAPI.search(
+					"+structureType:" + Structure.STRUCTURE_TYPE_FILEASSET + " +conFolder:" + parentFolder.getInode(),
+					-1, 0, null, user, respectFrontendRoles);
+		} catch (DotRuntimeException e) {
+			if (e.getCause() instanceof ConnectException) {
```

Comment:
this could be better to handle by ExceptionUtil.causedBy
The immediate cause could be a wrapper of the real one, should go in deep

Reply:
Should log too

Reply:
If you log, then we should use the `Logger.warnEveryAndDebug(final Class cl, final String message, final Throwable ex, final int warnEveryMillis) `

and write every 5000ms

Reply:
done https://github.com/dotCMS/core/pull/18225/commits/b3221ca6fdad2fc09e713c11a6b62f4339571aac#diff-3e213b868e011e1477445a17721c1926R141